ERCS/erc-7730.md

```diff
@@ -0,0 +1,442 @@
+
+* Function name or main message type is often a convoluted name and does not translate to a clear intent for the user
+* Fields in the data to sign are tied to primitive types only, but those can be displayed many different ways. For instance, integers can be displayed as percentages, dates, etc...
+* Some fields require additional metadata to by displayed correctly, for instance token amounts requires knowledge of the magnitude and the ticker, as well as where to find the token address itself to be correctly formatted.
+
+This specification intends to provide a simple, open standard format to provide wallets with the additional information required to properly format a structured data to sign for review by users.
+
+Providing this additional formatting information requires deep knowledge of the way a smartcontract or message is going to be used, it is expected that DApps developers will be the best placed to write such a file. The intent of an open standard is to write this file only once and have it work with most wallets supporting this standard.
+
+> Deployment model options:
+> - In each dapps repository (discoverability?)
+> - Foundation operated repository
+> - Ledger repository
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Simple example
+
+The following is a simple example of how to clear sign a `transfer` function call on an ERC20 contract.
+
+```json
+{
+    "$schema": "https://github.com/LedgerHQ/clear-signing-format-spec/blob/master/schemas/clear-signing-schema.json",
+
+    "context": {
+        "$id": "Tether USD",
+        "contract" : {
+            "chainId": 1,
+            "address": "0xdAC17F958D2ee523a2206206994597C13D831ec7",
+            "abi": "https://api.etherscan.io/api?module=contract&action=getabi&address=0xdac17f958d2ee523a2206206994597c13d831ec7"
+        }
+    }, 
+
+    "metadata": {
+        "owner": "Tether",
+        "info": {
+            "legalName": "Tether Limited",
+            "url": "https://tether.to/",
+            "lastUpdate": "2017-11-28T12:41:21Z"  
+        }
+    },
+
+    "display": {
+        "formats": {
+            "0xa9059cbb": {
+                "$id" : "transfer",
+                "intent": "Send",
+                "fields": {
+                    "_to" : {
+                        "label": "To",
+                        "format": "addressOrName"
+                    },
+                    "_value" : {
```

Comment:
Some contracts that implement the same standard like ERC-20 (and should ideally be using the same specification) are using different field names e.g. in USDT they are called `_to` and `_value` while in the DAI contract they are called `dst` and `wad`. Could this cause any issues when trying to parse the ABI? Also, is it expected that the there will be multiple specifications for the same function signature?

Reply:
That's why the 7730 file contains a reference of the ABI used. Field name should be the same as the ones in the ABI provided as reference. Since parameter name are not part of function signatures, that shouldn't prevent reusing the erc7730 file for other contracts that use different names for parameters